clustered/client/src/test/java/org/ehcache/clustered/client/internal/service/DefaultClusteringServiceTest.java

```diff
@@ -1933,6 +1938,63 @@ public void testGetServerStoreProxyReturnsStrongStore() throws Exception {
     assertThat(serverStoreProxy, instanceOf(StrongServerStoreProxy.class));
   }
 
+  @Test
+  public void testGetStateRepositoryWithinTwiceWithSameName() throws Exception {
+    ClusteringServiceConfiguration configuration =
+        new ClusteringServiceConfiguration(URI.create(CLUSTER_URI_BASE), true, new ServerSideConfiguration(Collections.<String, Pool>emptyMap()));
+    DefaultClusteringService service = new DefaultClusteringService(configuration);
+    PersistableResourceService.PersistenceSpaceIdentifier cacheIdentifier = service.getPersistenceSpaceIdentifier("myCache", null);
+    StateRepository repository1 = service.getStateRepositoryWithin(cacheIdentifier, "myRepo");
+    StateRepository repository2 = service.getStateRepositoryWithin(cacheIdentifier, "myRepo");
+    assertThat(repository1, sameInstance(repository2));
+  }
+
+  @Test
+  public void testGetStateRepositoryWithinTwiceWithSameNameDifferentPersistenceSpaceIdentifier() throws Exception {
+    ClusteringServiceConfiguration configuration =
+        new ClusteringServiceConfiguration(URI.create(CLUSTER_URI_BASE), true, new ServerSideConfiguration(Collections.<String, Pool>emptyMap()));
+    DefaultClusteringService service = new DefaultClusteringService(configuration);
+    PersistableResourceService.PersistenceSpaceIdentifier cacheIdentifier1 = service.getPersistenceSpaceIdentifier("myCache1", null);
+    PersistableResourceService.PersistenceSpaceIdentifier cacheIdentifier2 = service.getPersistenceSpaceIdentifier("myCache2", null);
+    StateRepository repository1 = service.getStateRepositoryWithin(cacheIdentifier1, "myRepo");
+    StateRepository repository2 = service.getStateRepositoryWithin(cacheIdentifier2, "myRepo");
+    assertThat(repository1, not(sameInstance(repository2)));
+  }
+
+  @Test(expected = CachePersistenceException.class)
```

Comment:
I really dislike using this construct when the test is more than a single line. Got bitten in the past by a test that was reporting the proper exception type but absolutely not the expected one.
Will merge the PR anyway but would appreciate a follow up PR fixing this and other similar tests in here.